python/pyxet/pyxet/url_parsing.py

```diff
@@ -24,229 +114,98 @@ def parse_url(url, default_domain='xethub.com', partial_remote=False, recursive_
     If partial_remote==True, allows [repo] to be optional. i.e. it will
     parse /user or xet://user
     """
+    url_info = url.split("://")
 
-    url = url.lstrip('/')
-    parse = urlparse(url)
-    if parse.scheme == '':
-        parse = parse._replace(scheme='xet')
+    if len(url_info) == 1: 
+        scheme = "xet"
+        url_path = url_info[0]
+    elif len(url_info) == 2:
+        scheme, url_path = url_info
```

Comment:
Looks like this `scheme` is never used, do we need to test if it is `xet`? The previous version seems to accept it as `https`, which I'm not sure why...

Reply:
I'm not sure here.  It was really hacky, but it seemed to just force everything to xet.  This function should only be called for xet URLs anyway.